ichnaea/locate/tests/test_provider.py

```diff
@@ -768,3 +768,40 @@ def test_redis_connection_failure_prevents_external_call(self):
             })
 
             self.assertFalse(location.found())
+
+    def test_single_cell_results_cached_preventing_external_call(self):
```

Comment:
I'd add at least one test to check that we don't cache if we get one cell + one or multiple wifis. The zero/zero or two cells cases are a bit stupid and I wouldn't test for them explicitly.